low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -443,6 +451,16 @@ def _generate_host_update(self):
             previous_check_time = int(memory_persistent_data['memory_usage_time_map'])
         else:
             previous_check_time = int(-1)
+        if memory_persistent_data\
+                .get('memory_fault_resolved_iterations') is not None:
+            fault_resolved_iters = int(
+                memory_persistent_data['memory_fault_resolved_iterations'])
+        else:
+            fault_resolved_iters = 0
+        try:
+            iteration_limit = int(self._high_memory_usage_wait_threshold/self._transmit_interval)

```

Comment:
Hi, @jjxsg , personally I feel following way easier, than calculating iteration,
```python
if time.time() - fault_resolved_time > wait_threshold:
    raise alert
```
But I don't know which way is better?

Reply:
Time based solution is what I used for fault alerts which is still there. In that case also we need to persist the fault resolved time in case fault comes again within the threshold waiting time. 
Iteration based approach handles the case when system or SSPL restarts within the waiting time interval for fault resolved alert. 

Let's say x is the time when fault is resolved and waiting time is 60 seconds which means fault resolved alert should be sent at time x+60.  Let's say reboot occurred at x+20 which took another 2 mins to restore the service in previous state. 
Now while checking whether to wait or send the alert, it will compare x+60 with (x+20+120) which is obviously greater, so it will send the alert immediately. But expected behavior is that it should wait for another 40 seconds before sending the alert.

Reply:
```
it will compare x+60 with (x+20+120) which is obviously greater, so it will send the alert immediately. 
But expected behavior is that it should wait for another 40 seconds before sending the alert.
```
Ohh really!! I was unaware of this expected behavior, I thought the service should be up for say 60s irrespective of whether sspl is observing it or not. 
Ideally, I think whether sspl is watching the service or not should not matter, we can directly check the uptime of the service and if the service is up for more than 60 sec then we can raise fault_resolved. This way we won't have to save time or count to cache as well.

Reply:
Well, this is true for a service because its state doesn't fluctuate frequently. But CPU and Memory usage fluctuate, so we need to check the usage in real time rather than depending on past values.

Reply:
Ohh sorry this is related to cpu and memory usage, my bad.😅 Then that makes sense, Thanks the patient explanation @jjxsg 👍